crates/chia-sdk-driver/src/primitives/cat.rs

```diff
@@ -761,6 +778,59 @@ mod tests {
         Ok(())
     }
 
+    #[rstest]
+    fn test_cat_tail_reveal(
+        #[values(0, 1, 2)] tail_index: usize,
+        #[values(true, false)] melt: bool,
+    ) -> anyhow::Result<()> {
+        let mut sim = Simulator::new();
+        let ctx = &mut SpendContext::new();
+
+        let alice = sim.bls(15000);
+        let alice_p2 = StandardLayer::new(alice.pk);
+        let hint = ctx.hint(alice.puzzle_hash)?;
+
+        let conditions = Conditions::new()
+            .create_coin(alice.puzzle_hash, 3000, hint)
+            .create_coin(alice.puzzle_hash, 6000, hint)
+            .create_coin(alice.puzzle_hash, 1000, hint);
+
+        let (issue_cat, cats) =
+            Cat::issue_with_key(ctx, alice.coin.coin_id(), alice.pk, 10000, conditions)?;
+
+        alice_p2.spend(ctx, alice.coin, issue_cat)?;
+
+        let tail = ctx.curry(EverythingWithSignatureTailArgs::new(alice.pk))?;
+
+        let cat_spends = cats
+            .into_iter()
+            .enumerate()
+            .map(|(i, cat)| {
+                let mut conditions = Conditions::new();
+
+                // Add the TAIL reveal to the second spend, to ensure the order doesn't matter
```

Comment:
That comment is not quite accurate anymore, now that you added the test parameters